lpeg/io_modules/lpeg/sub_decoder_util.lua

```diff
@@ -164,6 +184,20 @@ local function grammar_pick_fn(sd, nomatch_action)
 end
 
 
+local mod_ref = "([^#]+)#(.+)"
+local function get_module_ref(s)
+    local mname, ref = string.match(s, mod_ref)
+    local mr
+    if mname then
+        mr = require(mname)[ref]
+    else
+        local m = require(s)
+        mr = m.grammar or m.syslog_grammar
+    end
```

Comment:
Should we validate mr is valid here? e.g., will an error here be indicated immediately or would it have to wait for the first attempt to use the decoder?

Reply:
Actually, disregard on this one -- it looks like we check the result type later on after it's called